pkg/policy/api/selector.go

```diff
@@ -34,16 +34,25 @@ var log = logging.DefaultLogger.WithField(logfields.LogSubsys, "policy-api")
 type EndpointSelector struct {
 	*slim_metav1.LabelSelector
 
-	// requirements provides a cache for a k8s-friendly format of the
+	// TODO: The following fields were exported to stop govet warnings. The
+	// govet warnings were because the CRD generation tool needs every struct
+	// field that's within a CRD, to have a json tag. JSON tags cannot be
+	// applied to unexported fields, hence this change. Refactor these fields
+	// out of this struct. GH issue:
+	// https://github.com/cilium/cilium/issues/12697. Once
+	// https://go-review.googlesource.com/c/tools/+/245857 is merged, this
+	// would no longer be required.
+
+	// Requirements provides a cache for a k8s-friendly format of the
 	// LabelSelector, which allows more efficient matching in Matches().
 	//
 	// Kept as a pointer to allow EndpointSelector to be used as a map key.
-	requirements *k8sLbls.Requirements `json:"-"`
+	Requirements *k8sLbls.Requirements `json:"-"`
 
 	// cachedString is the cached representation of the LabelSelector for this
 	// EndpointSelector. It is populated when EndpointSelectors are created
 	// via `NewESFromMatchRequirements`. It is immutable after its creation.
-	cachedLabelSelectorString string
+	CachedLabelSelectorString string `json:"-"`
```

Comment:
The addition of the JSON tag should have been done in the previous commit.

Reply:
Fixed